src/pkg/packager/sources/oci.go

```diff
@@ -40,11 +40,17 @@ func (s *OCISource) LoadPackage(dst *layout.PackagePaths, unarchiveAll bool) (er
 
 	message.Debugf("Loading package from %q", s.PackageSource)
 
-	optionalComponents := helpers.StringToSlice(s.OptionalComponents)
-
 	// pull only needed layers if --confirm is set
 	if config.CommonOptions.Confirm {
-		layersToPull, err = s.LayersFromRequestedComponents(optionalComponents)
+		pkg, err = s.FetchZarfYAML()
+		if err != nil {
+			return err
+		}
+		requested, err := interactive.GetSelectedComponents(s.OptionalComponents, pkg.Components)
```

Comment:
we should use `ForIncludedComponents` instead of `GetSelectedComponents` for remove and mirror operations

Reply:
remove does not need this performed, as it only calls `s.LoadPackageMetadata`